pkg/bootstrap/docker/openshift/servicecatalog.go

```diff
@@ -0,0 +1,121 @@
+package openshift
+
+import (
+	"bytes"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/golang/glog"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/openshift/origin/pkg/bootstrap/docker/errors"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+const (
+	catalogNamespace        = "service-catalog"
+	catalogService          = "service-catalog"
+	catalogTemplate         = "service-catalog"
+	ServiceCatalogServiceIP = "172.30.1.2"
+)
+
+// InstallServiceCatalog checks whether the service catalog is installed and installs it if not already installed
+func (h *Helper) InstallServiceCatalog(f *clientcmd.Factory, publicMaster, catalogHost string) error {
+	osClient, kubeClient, err := f.Clients()
+	if err != nil {
+		return errors.NewError("cannot obtain API clients").WithCause(err).WithDetails(h.OriginLog())
+	}
+
+	_, err = kubeClient.Core().Namespaces().Get(catalogNamespace, metav1.GetOptions{})
+	if err == nil {
+		// If there's no error, the catalog namespace already exists and we won't initialize it
+		return nil
+	}
+
+	// Create catalog namespace
+	out := &bytes.Buffer{}
+	err = CreateProject(f, catalogNamespace, "", "", "oc", out)
+	if err != nil {
+		return errors.NewError("cannot create service catalog project").WithCause(err).WithDetails(out.String())
+	}
+
+	// Instantiate service catalog
+	params := map[string]string{
+		"SERVICE_CATALOG_SERVICE_IP":     ServiceCatalogServiceIP,
+		"SERVICE_CATALOG_ROUTE_HOSTNAME": catalogHost,
+		"CORS_ALLOWED_ORIGIN":            publicMaster,
+	}
+	glog.V(2).Infof("instantiating service catalog template")
+
+	err = instantiateTemplate(osClient, clientcmd.ResourceMapper(f), "openshift", catalogTemplate, catalogNamespace, params)
+	if err != nil {
+		return errors.NewError("cannot instantiate service catalog template").WithCause(err)
+	}
+
+	err = wait.Poll(5*time.Second, 600*time.Second, func() (bool, error) {
+		glog.V(2).Infof("polling for service catalog api server endpoint availability")
+		deployment, err := kubeClient.Extensions().Deployments(catalogNamespace).Get("apiserver", metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		if deployment.Status.AvailableReplicas > 0 {
+			return true, nil
+		}
+		return false, nil
+	})
+	if err != nil {
+		return errors.NewError(fmt.Sprintf("failed to start the service catalog apiserver: %v", err))
+	}
+
+	insecureCli := http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		},
+	}
+	register := `{
```

Comment:
it may or may not stay there.  the content in examples/service-catalog is a bit of a WIP.  it's handy for bootstrapping the service catalog/template broker manually.